packages/desktop-client/src/components/reports/util.ts

```diff
@@ -13,23 +17,23 @@ export async function runAll(queries, cb) {
   cb(data);
 }
 
-export function index(data, field, mapper) {
+export function index(data, field: string, mapper?: (input) => unknown) {
   const result = {};
   data.forEach(item => {
     result[mapper ? mapper(item[field]) : item[field]] = item;
   });
   return result;
 }
 
-export function indexStack(data, fieldName, field) {
+export function indexStack(data, fieldName: string, field: string) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this one and also `indexCashFlow` (below) could be improved with generics too.

Reply:
I tried implementing it for indexCashFlow but it is not taking generic types quite well